cmd/env.go

```diff
@@ -361,3 +322,32 @@ func commonEnvFlags(flags *pflag.FlagSet) (server, namespace, context string, er
 
 	return server, namespace, context, nil
 }
+
+func resolveEnvFlags(flags *pflag.FlagSet) (server, namespace string, err error) {
+	server, ns, ctx, err := commonEnvFlags(flags)
+	if err != nil {
+		return "", "", err
+	}
+
+	if len(server) == 0 {
+		// server is not provided -- use the context.
+		if len(ctx) != 0 {
+			server, ns, err = resolveContextFor(ctx)
+		} else {
+			// use the default context
+			server, ns, err = resolveContext()
+		}
+
+		if err != nil {
+			return "", "", err
+		}
+	}
+
+	if len(ns) == 0 {
+		namespace = defaultNamespace
+	} else {
+		namespace = ns
+	}
+
+	return
```

Comment:
Instead of the naked return, do a `return server, ns, nil` and change the conditional above to 

```go
if ns == "" {
  ns = defaultNamespace
}